### ISSUE TITLE
Inferred properties aren't detected in 3.0.3 documents

### ISSUE THREAD

**Issue by MahdiBM** [apple/swift-openapi-generator#437]
```yml
components:
  schemas:
    MyType:
      type: object
      properties:
        someName:
          type: string
      required:
        - anotherName /// <--- non-existent required field
```
The generator generates both `someName` and `anotherName` fields. The `anotherName` field will be of type `OpenAPIRuntime.OpenAPIValueContainer`. `someName` will be of the correct type.

OpenAPI should either error or ignore such stuff in the `required` values.

**Comment by MahdiBM** [comment]
It is `alpha.1`. It does not emit any diagnostics.

**Comment by czechboy0** [comment]
And this test is passing, so this is really interesting: https://github.com/apple/swift-openapi-generator/blob/main/Tests/OpenAPIGeneratorReferenceTests/SnippetBasedReferenceTests.swift#L216

Your case must be somehow different to the one we have in our tests, can you provide more details?

**Comment by MahdiBM** [comment]
@czechboy0 Here a full document that reproduces this:
```yml
openapi: 3.0.3
info:
  title: API
  description: |-
    description
  termsOfService: http://some.thing/terms/
  contact:
    email: apiteam@some.thing
  license:
    name: Apache 2.0
    url: http://www.apache.org/licenses/LICENSE-2.0.html
  version: 1.0.0
externalDocs:
  description: Find out more about API
  url: http://some.thing
servers:
  - url: https://some.thing
paths:
  /reports/id/{id}:
    get:
      summary: Get an existing report by ID
      description: Get an existing report by ID
      operationId: getReportById
      parameters:
        - name: id
          in: path
          description: ID of report to return
          required: true
          schema:
            type: integer
      responses:
        200:
          description: Report exists and is returned
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Report'
        400:
          description: Invalid ID supplied
      security:
        - bearerAuth: []
components:
  securitySchemes:
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: JWT
  schemas:
    Report:
      type: object
      properties:
        file_type:
          type: string
          enum: [a]
        report_category:
          type: string
          enum: [b]
      required:
        - type
        - category
```

**Comment by czechboy0** [comment]
Ok found it, seems it works in `3.1.0` documents fine, but not in `3.0.3`, for some reason. We use OpenAPIKit for this, so tagging @mattpolzin for help, I suspect the fix will have to come there.

**Comment by czechboy0** [comment]
Wow that was fast, thank you @mattpolzin! 👏

Before we close this issue, @MahdiBM can you update to latest dependencies in your checkout, regenerate, and verify the issue is fixed, before we close it here? 🙏

**Comment by MahdiBM** [comment]
Nevermind, i see the warning.

**Comment by czechboy0** [comment]
Thanks @MahdiBM for filing this and helping us get it addressed before 1.0 🙏 And thank you @mattpolzin for the speedy fix, as always! 👏
